modules/channel-messenger/src/backend/messenger.ts

```diff
@@ -0,0 +1,253 @@
+
+      if (webhookEvent.message) {
+        await this._sendEvent(botId, senderId, webhookEvent.message, { type: 'message' })
+      } else if (webhookEvent.postback) {
+        await this._sendEvent(botId, senderId, { text: webhookEvent.postback.payload }, { type: 'callback' })
+      }
+    }
+
+    res.status(200).send('EVENT_RECEIVED')
+  }
+
+  private async _sendEvent(botId: string, senderId: string, message, args: { type: string }) {
+    this.bp.events.sendEvent(
+      this.bp.IO.Event({
+        botId,
+        channel: 'messenger',
+        direction: 'incoming',
+        payload: message,
+        preview: message.text,
+        target: senderId,
+        ...args
+      })
+    )
+  }
+
+  private async _setupWebhook(req, res) {
+    const mode = req.query['hub.mode']
+    const token = req.query['hub.verify_token']
+    const challenge = req.query['hub.challenge']
+    const botId = req.params.botId
+
+    const client = this.getMessengerClient(botId)
+    const config = await client.getConfig()
+
+    if (mode && token && mode === 'subscribe' && token === config.verifyToken) {
+      this.bp.logger.forBot(botId).debug('Webhook Verified.')
+      res.status(200).send(challenge)
+    } else {
+      res.sendStatus(403)
+    }
+
+    await client.setupGreeting()
+    await client.setupGetStarted()
+    await client.setupPersistentMenu()
+  }
+
+  private async _handleOutgoingEvent(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    if (event.channel !== 'messenger') {
+      return next()
+    }
+
+    const messageType = event.type === 'default' ? 'text' : event.type
+    const messenger = this.getMessengerClient(event.botId)
+
+    if (!_.includes(outgoingTypes, messageType)) {
+      return next(new Error('Unsupported event type: ' + event.type))
+    }
+
+    if (messageType === 'typing') {
+      await messenger.sendAction(event.target, 'typing_on')
+    } else if (messageType === 'text' || messageType === 'carousel') {
+      await messenger.sendTextMessage(event.target, event.payload)
+    } else {
+      // TODO We don't support sending files, location requests (and probably more) yet
+      throw new Error(`Message type "${messageType}" not implemented yet`)
+    }
+
+    next(undefined, false)
+  }
+}
+
+export class MessengerClient {
+  private config: Config
+
+  constructor(private botId: string, private bp: typeof sdk, private http: AxiosInstance) {}
+
+  async getConfig(): Promise<Config> {
+    if (this.config) {
+      return this.config
+    }
+
+    const config = (await this.bp.config.getModuleConfigForBot('channel-messenger', this.botId)) as Config
+    if (!config) {
+      throw new Error(`Could not find channel-messenger.json config file for ${this.botId}.`)
+    }
+
+    return config
+  }
+
+  async setupGetStarted(): Promise<void> {
+    const config = await this.getConfig()
+    if (!config.getStarted) {
+      return
+    }
+
+    const body = {
```

Comment:
It would be nice if you could use the bot details data to setup the getStarted config and make it overridable by the module configs